src/app/core/env-config.service.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { LocationStrategy } from '@angular/common';
+import { EMPTY, Observable, tap } from 'rxjs';
+import { environment } from '../../environments/environment';
+
+export interface EnvConfig {}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class EnvConfigService {
+  static readonly CONFIGURATION_FILE_NAME: string = 'envConfig.json';
```

Comment:
Shouldn't we have a default `envConfig.json`? Won't it throw 404 error in production build?

Reply:
This file should be delivered when deploying on different environments.
I think app should not be able to start when there is no config file as it's crucial for app to work properly.